### PR TITLE
add stolpersteine to existing memorials-theme

### PR DESCRIPTION
**Pull request by hejjoe**
Opening a pull request on MapComplete
=====================================

Hey! Thanks for opening a pull request on Mapcomplete! This probably means you want to add a new theme - if so, please
follow the checklist below. If this pull request is for some other issue, please ignore the template.

Adding your new theme
---------------------

Thanks for taking the time to create your theme and to add it to the main repository!

To making merging smooth, please make sure that each of the following conditions is met:

- [X] The codebase is GPL-licensed. By opening a pull request, the new theme will be GPL too
- [X] All images are included in the pull request and no images are loaded from an external service (e.g. Wikipedia)
- [X] The [guidelines on how to make your own theme](https://github.com/pietervdvn/MapComplete/blob/master/Docs/Making_Your_Own_Theme.md)
  are read and followed

### CODE REVIEW

File: assets/layers/stolperstein/stolperstein.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
+{
+  "id": "stolperstein",
+  "name": {
+    "en": "stolperstein",
+    "de": "Stolperstein"
+  },
+  "description": {
+    "en": "Stolpersteine (or stumbling stones) indicate where a person lived. They are intended to commemorate the fate of people who were persecuted, murdered, deported, expelled or driven to suicide during the National Socialist era. These stumbling blocks can be marked on this map and a Wikipedialink or Wikidata can be created.",
```
Comment (jaller94) [inline assets/layers/stolperstein/stolperstein.json:8]:
```suggestion
    "en": "Stolpersteine indicate where a person lived. They are intended to commemorate the fate of people who were persecuted, murdered, deported, expelled or driven to suicide during the National Socialist era. These Stolpersteine can be marked on this map and a Wikipedia link or Wikidata can be created.",
```

File: assets/layers/stolperstein/stolperstein.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    "de": "Stolpersteine zeigen an, wo ein Mensch gewohnt hat. Sie sollen an das Schicksal der Menschen erinnert werden, die in der Zeit des Nationalsozialismus verfolgt, ermordet, deportiert, vertrieben oder in den Suizid getrieben wurden. Auf dieser Karte können diese Stolpersteine aufgezeichnet werden und ein Wikipedialink bzw. Wikidata erstellt werden"
```
Comment (jaller94) [inline assets/layers/stolperstein/stolperstein.json:9]:
```suggestion
    "de": "Stolpersteine zeigen an, wo ein Mensch gewohnt hat. Sie sollen an das Schicksal der Menschen erinnert werden, die in der Zeit des Nationalsozialismus verfolgt, ermordet, deportiert, vertrieben oder in den Suizid getrieben wurden. Auf dieser Karte können diese Stolpersteine aufgezeichnet werden und ein Wikipedia-Link bzw. Wikidata erstellt werden."
```

File: assets/layers/stolperstein/stolperstein.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      }
+    }
+  ],
+  "filter": [
+    {
+      "id": "has_wikidata",
+      "options": [
+        {
+          "question": {
+            "en": "With or without wikidata",
```
Comment (jaller94) [inline assets/layers/stolperstein/stolperstein.json:91]:
```suggestion
            "en": "With or without Wikidata",
```

File: assets/layers/stolperstein/stolperstein.json [hunk 4]
```diff
@@ -0,0 +1,117 @@
+{
+  "id": "stolperstein",
+  "name": {
+    "en": "stolperstein",
```
Comment (jaller94) [inline assets/layers/stolperstein/stolperstein.json:4]:
```suggestion
    "en": "Stolperstein",
```

**Comment by pietervdvn** [comment]
I have to dissapoint you all a little bit, as stolpersteine are already included in the `memorials`-layer. The first question specifies what type of memorial is added, of which stumbling stones is one of the options.

As you have created this pull request, stumbling stones would be shown in the `memorial`-layer and not the `stolperstein`-layer as you've created them.

Also note that the memorial layer already _has_ support for `wikidata`, `subject:wikidata`, `subject` and `inscription`. Using the `name`-field is semantically incorrect to store who is remembered (even though this is a common error).

As this PR doesn't add new functionality (but only increases code size), I will not be merging this PR.

Know however that I really appreciate the time and affort that you put into this. I'm always happy if people try to create new themes or want to improve MapComplete, so I hope to welcome you again sometime soon!

**Comment by pietervdvn** [comment]
Another question: as you've gone through the effort to create this PR, this also indicates that you had a certain usecase in mind which goes unfulffilled - you'll probably want a map showing all stolpersteine.

This can be achieved by either adding a filter to the "memorials"-map to filter based on type of memorial. Another option is to have a extra thematic map, which reuses the `memorials`-layer but filters it so that it only shows stolpersteine. This can be done with the `override`-constructions.

This avoids code duplication (and updates to the `memorial`-layer would automatically be reflected as well) but serves the usecase too.

**Comment by hejjoe** [comment]
The idea was to add a "stolperstein with or without wikidata"-filter to [memorials theme](https://mapcomplete.org/memorials.html), just like the filter on graves. So I took the [graves.json](https://github.com/pietervdvn/MapComplete/blob/master/assets/layers/grave/grave.json) and transformed into a stolperstein.json. Maybe this goal of better filtering could be achieved in a different way. I was not sure how to add and combine more filters.
grafik

**Pull request closed without merging**